meilisearch-http/src/helpers/meilisearch.rs

```diff
@@ -171,39 +184,15 @@ impl<'a> SearchBuilder<'a> {
             None => self.index.query_builder(),
         };
 
-        if let Some(filters) = &self.filters {
-            let mut split = filters.split(':');
-            match (split.next(), split.next()) {
-                (Some(_), None) | (Some(_), Some("")) => return Err(Error::MissingFilterValue),
-                (Some(attr), Some(value)) => {
-                    let ref_reader = reader;
-                    let ref_index = &self.index;
-                    let value = value.trim().to_lowercase();
-
-                    let attr = match schema.id(attr) {
-                        Some(attr) => attr,
-                        None => return Err(Error::UnknownFilteredAttribute),
-                    };
-
-                    query_builder.with_filter(move |id| {
-                        let attr = attr;
-                        let index = ref_index;
-                        let reader = ref_reader;
-
-                        match index.document_attribute::<Value>(reader, id, attr) {
-                            Ok(Some(Value::String(s))) => s.to_lowercase() == value,
-                            Ok(Some(Value::Bool(b))) => {
-                                (value == "true" && b) || (value == "false" && !b)
-                            }
-                            Ok(Some(Value::Array(a))) => {
-                                a.into_iter().any(|s| s.as_str() == Some(&value))
-                            }
-                            _ => false,
-                        }
-                    });
-                }
-                (_, _) => (),
-            }
+        if let Some(filter_expression) = &self.filters {
+            let filter = Filter::parse(filter_expression, &schema)?;
+            query_builder.with_filter(move |id| {
+                let index = &self.index;
+                let reader = &reader;
+                let filter = &filter;
+                filter.test(reader, index, id).expect("filter error")
```

Comment:
```suggestion
                filter.test(reader, index, id).unwrap_or(false)
```

I know that it is not ideal but I would prefer avoid crashing a search if one filter fail.

Reply:
Oopsie, I thought I had changed it to warn, fixing it.